src/LodeRunner/src/Services/LodeRunnerService.cs

```diff
@@ -293,12 +336,20 @@ private async Task<int> StartAndWait()
                     {
                         foreach (var testRun in testRuns)
                         {
-                            this.StatusUpdate(null, new ClientStatusEventArgs(ClientStatusType.Testing, $"Received new TestRun ({testRun.Id})", this.cancellationTokenSource));
-
-                            // Only execute TestRuns scheduled to run before the next minute
-                            if (testRun.StartTime < DateTime.UtcNow.AddMinutes(1))
+                            // skip tests that have been completed but not yet updated with results in cosmos
+                            if (!this.pendingTestRuns.Contains(testRun.Id))
                             {
-                                await this.ExecuteNewTestRunAsync(testRun);
+                                // Only execute TestRuns scheduled to run before the next minute
+                                if (testRun.StartTime < DateTime.UtcNow.AddMinutes(1))
+                                {
+                                    this.StatusUpdate(null, new ClientStatusEventArgs(ClientStatusType.Testing, $"Received new TestRun ({testRun.Id})", this.cancellationTokenSource));
+
+                                    // Only execute TestRuns scheduled to run before the next minute
+                                    if (testRun.StartTime < DateTime.UtcNow.AddMinutes(1))
```

Comment:
Do we need to check this twice? (ref line 343)

Reply:
No :) this one snuck in after merge conflicts, thanks!